engine/src/main/java/org/hibernate/validator/internal/engine/ConfigurationImpl.java

```diff
@@ -304,7 +315,17 @@ public MethodValidationConfiguration getMethodValidationConfiguration() {
 
 	@Override
 	public final DefaultConstraintMapping createConstraintMapping() {
-		return new DefaultConstraintMapping();
+		PropertyFilter propertyFilterToUse = null;
+		if ( propertyFilter == null ) {
+			propertyFilterToUse = new DefaultPropertyFilter();
+			LOG.creatingConstraintMappingPriorToBuildingFactoryWithoutDefinedFilter();
+		}
+		else {
+			propertyFilterToUse = propertyFilter;
+		}
+		LOG.creatingConstraintMappingPriorToBuildingFactory( propertyFilterToUse.getClass() );
+
+		return new DefaultConstraintMapping( propertyFilterToUse );
```

Comment:
@gsmet even though we expose this "property selection functionality" only through the `HibernateValidatorConfiguration` we still would have a "problem", as it's at the same level where we add constraint mappings so if user adds mappings and then change the `propertyFilter` it most likely would not end good. I've added a few warnings here, as I don't see a nice solution for now. Unless we want to just collect the info here and delay all the work until later time, when building the factory. 
Other than that I think you can give it a first look, when you have time for it :smiley:.

Reply:
Hmmm, it's the second time we hit this wall (we already had this issue when we discussed resolving the parameter names once and for all - it's a bit different here though as we don't have the `HIbernateValidatorContext` issue). Taking a step back, I think it was a pretty bad decision to allow to define the mappings during the configuration phase. It should have been a separate step.

Wondering if we should revisit this idea of postponing the mappings creation to the factory building phase.

But it's something we can do after getting this patch in, if we think it makes sense.